test/app-luatest/shutdown_test.lua

```diff
@@ -200,3 +202,72 @@ g_standby.test_shutdown_during_hot_standby = function(cg)
     end)
     test_no_hang_on_shutdown(cg.standby)
 end
+
+-- Luatest server currently does not allow to check process exit code.
+local g_crash = t.group('crash')
+
+g_crash.before_each(function(cg)
+    local id = ('%s-%s'):format('server', utils.generate_id())
+    cg.workdir = fio.pathjoin(server.vardir, id)
+    fio.mkdir(cg.workdir)
+end)
+
+g_crash.after_each(function(cg)
+    if cg.handle ~= nil then
+        cg.handle:close()
+    end
+    cg.handle = nil
+end)
+
+local tarantool = arg[-1]
+
+-- Test shutdown does not hang due to memtx space snapshot in progress.
+g_crash.test_shutdown_during_snapshot_on_signal = function(cg)
+    t.tarantool.skip_if_not_debug()
+    local script = [[
+        local fiber = require('fiber')
+        local fio = require('fio')
+        local log = require('log')
+
+        local workdir = os.getenv('TARANTOOL_WORKDIR')
+        fio.chdir(workdir)
+        local logfile = fio.pathjoin(workdir, 'server.log')
+        box.cfg{log = logfile}
+        box.schema.create_space('test')
+        box.space.test:create_index('pk')
+        box.begin()
+        for i=1,10000 do
+            box.space.test:insert{i}
+        end
+        box.commit()
+        box.error.injection.set('ERRINJ_SNAP_WRITE_TIMEOUT', 0.01)
+        log.info('server startup finished')
+    ]]
```

Comment:
Please consider using `justrun` for this.

Reply:
Unfortunately I cannot send `SIGUSR1` for Tarantool run with `justrun` easily.